### PR TITLE
bootstrap: fallback to default known_hosts

### PR DESCRIPTION
**Pull request by aryan9600**
Use default known_hosts and ssh configuration when no private key file is provided while bootstraping using ssh.

Fixes: #3358 
Signed-off-by: Sanskar Jaiswal <jaiswalsanskar078@gmail.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,17 +54,17 @@ $(EMBEDDED_MANIFESTS_TARGET): $(call rwildcard,manifests/,*.yaml *.json)
...
+	CGO_ENABLED=1 go build -ldflags="-s -w -X main.VERSION=$(VERSION)" -o ./bin/flux ./cmd/flux
 
 build-dev: $(EMBEDDED_MANIFESTS_TARGET)
-	CGO_ENABLED=0 go build -ldflags="-s -w -X main.VERSION=$(DEV_VERSION)" -o ./bin/flux ./cmd/flux
+	CGO_ENABLED=1 go build -ldflags="-s -w -X main.VERSION=$(DEV_VERSION)" -o ./bin/flux ./cmd/flux
 
 .PHONY: install
 install:
-	CGO_ENABLED=0 go install ./cmd/flux
+	CGO_ENABLED=1 go install ./cmd/flux
```
Comment (pjbgf) [inline Makefile:64]: As we as still flushing this through the repos, add a replace instead:
```suggestion
	replace github.com/pjbgf/sha1cd => github.com/pjbgf/sha1cd v0.2.3
```

File: go.mod [hunk 2]
```diff
@@ -2,21 +2,23 @@ module github.com/fluxcd/flux2
 
 go 1.18
 
+replace github.com/pjbgf/sha1cd => github.com/pjbgf/sha1cd v0.2.3
```
[the remark this answers is not in the document]
Comment (pjbgf) [reply go.mod:5]: @stefanprodan This was a temporary fix for the `CGO_ENABLED` issue. Now that this was already fixed for both other repos, the replace is no longer needed.

@aryan9600 please update both `go-git` and `pkg/git/gogit` to the latest tagged versions.

File: go.mod [hunk 3]
```diff
@@ -8,15 +8,15 @@ require (
 	github.com/cyphar/filepath-securejoin v0.2.3
 	github.com/distribution/distribution/v3 v3.0.0-20221119093643-85d4039064cc
 	github.com/fluxcd/go-git-providers v0.11.0
-	github.com/fluxcd/go-git/v5 v5.0.0-20221104190732-329fd6659b10
+	github.com/fluxcd/go-git/v5 v5.0.0-20221206140629-ec778c2c37df
```
Comment (stefanprodan) [inline go.mod:11]: We need to keep this in-sync with https://github.com/fluxcd/flux2/blob/main/tests/azure/go.mod#L7

**Review by pjbgf — APPROVED** [review]
LGTM

Thanks @aryan9600! :bow:

**Pull request merged**